### ISSUE TITLE
Unhandled Exception

### ISSUE THREAD

**Issue by Abulph** [Codeusa/Borderless-Gaming#220]
Whenever I try to use Borderless Gaming, this comes up:
![issue](https://cloud.githubusercontent.com/assets/16374168/11919461/71cdca50-a721-11e5-983b-c1da7e071cbc.PNG)

I found the log and this is what is has to say:

{
  "ClassName": "System.AggregateException",
  "Message": "A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread.",
  "Data": null,
  "InnerException": {
    "ClassName": "System.IO.FileNotFoundException",
    "Message": "Could not load file or assembly 'System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.",
    "Data": null,
    "InnerException": null,
    "HelpURL": null,
    "StackTraceString": "   at BorderlessGaming.BorderlessGaming.DoMainWork()\r\n   at System.Threading.Tasks.Task.Execute()",
    "RemoteStackTraceString": null,
    "RemoteStackIndex": 0,
    "ExceptionMethod": "8\nDoMainWork\nBorderlessGaming, Version=8.4.33315.1328, Culture=neutral, PublicKeyToken=null\nBorderlessGaming.BorderlessGaming\nVoid DoMainWork()",
    "HResult": -2147024894,
    "Source": "BorderlessGaming",
    "WatsonBuckets": null,
    "FileNotFound_FileName": "System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
    "FileNotFound_FusionLog": "WRN: Assembly binding logging is turned OFF.\r\nTo enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog](DWORD) to 1.\r\nNote: There is some performance penalty associated with assembly bind failure logging.\r\nTo turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].\r\n"
  },
  "HelpURL": null,
  "StackTraceString": "   at System.Threading.Tasks.TaskExceptionHolder.Finalize()",
  "RemoteStackTraceString": null,
  "RemoteStackIndex": 0,
  "ExceptionMethod": "8\nFinalize\nmscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089\nSystem.Threading.Tasks.TaskExceptionHolder\nVoid Finalize()",
  "HResult": -2146233088,
  "Source": "mscorlib",
  "WatsonBuckets": null,
  "InnerExceptions": [
    {
      "ClassName": "System.IO.FileNotFoundException",
      "Message": "Could not load file or assembly 'System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.",
      "Data": null,
      "InnerException": null,
      "HelpURL": null,
      "StackTraceString": "   at BorderlessGaming.BorderlessGaming.DoMainWork()\r\n   at System.Threading.Tasks.Task.Execute()",
      "RemoteStackTraceString": null,
      "RemoteStackIndex": 0,
      "ExceptionMethod": "8\nDoMainWork\nBorderlessGaming, Version=8.4.33315.1328, Culture=neutral, PublicKeyToken=null\nBorderlessGaming.BorderlessGaming\nVoid DoMainWork()",
      "HResult": -2147024894,
      "Source": "BorderlessGaming",
      "WatsonBuckets": null,
      "FileNotFound_FileName": "System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
      "FileNotFound_FusionLog": "WRN: Assembly binding logging is turned OFF.\r\nTo enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog](DWORD) to 1.\r\nNote: There is some performance penalty associated with assembly bind failure logging.\r\nTo turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].\r\n"
    }
  ]
}

{
  "ClassName": "System.AggregateException",
  "Message": "A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread.",
  "Data": null,
  "InnerException": {
    "ClassName": "System.IO.FileNotFoundException",
    "Message": "Could not load file or assembly 'System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.",
    "Data": null,
    "InnerException": null,
    "HelpURL": null,
    "StackTraceString": "   at BorderlessGaming.BorderlessGaming.DoMainWork()\r\n   at System.Threading.Tasks.Task.Execute()",
    "RemoteStackTraceString": null,
    "RemoteStackIndex": 0,
    "ExceptionMethod": "8\nDoMainWork\nBorderlessGaming, Version=8.4.33315.1328, Culture=neutral, PublicKeyToken=null\nBorderlessGaming.BorderlessGaming\nVoid DoMainWork()",
    "HResult": -2147024894,
    "Source": "BorderlessGaming",
    "WatsonBuckets": null,
    "FileNotFound_FileName": "System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
    "FileNotFound_FusionLog": "WRN: Assembly binding logging is turned OFF.\r\nTo enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog](DWORD) to 1.\r\nNote: There is some performance penalty associated with assembly bind failure logging.\r\nTo turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].\r\n"
  },
  "HelpURL": null,
  "StackTraceString": "   at System.Threading.Tasks.TaskExceptionHolder.Finalize()",
  "RemoteStackTraceString": null,
  "RemoteStackIndex": 0,
  "ExceptionMethod": "8\nFinalize\nmscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089\nSystem.Threading.Tasks.TaskExceptionHolder\nVoid Finalize()",
  "HResult": -2146233088,
  "Source": "mscorlib",
  "WatsonBuckets": null,
  "InnerExceptions": [
    {
      "ClassName": "System.IO.FileNotFoundException",
      "Message": "Could not load file or assembly 'System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.",
      "Data": null,
      "InnerException": null,
      "HelpURL": null,
      "StackTraceString": "   at BorderlessGaming.BorderlessGaming.DoMainWork()\r\n   at System.Threading.Tasks.Task.Execute()",
      "RemoteStackTraceString": null,
      "RemoteStackIndex": 0,
      "ExceptionMethod": "8\nDoMainWork\nBorderlessGaming, Version=8.4.33315.1328, Culture=neutral, PublicKeyToken=null\nBorderlessGaming.BorderlessGaming\nVoid DoMainWork()",
      "HResult": -2147024894,
      "Source": "BorderlessGaming",
      "WatsonBuckets": null,
      "FileNotFound_FileName": "System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
      "FileNotFound_FusionLog": "WRN: Assembly binding logging is turned OFF.\r\nTo enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog](DWORD) to 1.\r\nNote: There is some performance penalty associated with assembly bind failure logging.\r\nTo turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].\r\n"
    }
  ]
}

{
  "ClassName": "System.AggregateException",
  "Message": "A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread.",
  "Data": null,
  "InnerException": {
    "ClassName": "System.IO.FileNotFoundException",
    "Message": "Could not load file or assembly 'System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.",
    "Data": null,
    "InnerException": null,
    "HelpURL": null,
    "StackTraceString": "   at BorderlessGaming.BorderlessGaming.DoMainWork()\r\n   at System.Threading.Tasks.Task.Execute()",
    "RemoteStackTraceString": null,
    "RemoteStackIndex": 0,
    "ExceptionMethod": "8\nDoMainWork\nBorderlessGaming, Version=8.4.33315.1328, Culture=neutral, PublicKeyToken=null\nBorderlessGaming.BorderlessGaming\nVoid DoMainWork()",
    "HResult": -2147024894,
    "Source": "BorderlessGaming",
    "WatsonBuckets": null,
    "FileNotFound_FileName": "System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
    "FileNotFound_FusionLog": "WRN: Assembly binding logging is turned OFF.\r\nTo enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog](DWORD) to 1.\r\nNote: There is some performance penalty associated with assembly bind failure logging.\r\nTo turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].\r\n"
  },
  "HelpURL": null,
  "StackTraceString": "   at System.Threading.Tasks.TaskExceptionHolder.Finalize()",
  "RemoteStackTraceString": null,
  "RemoteStackIndex": 0,
  "ExceptionMethod": "8\nFinalize\nmscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089\nSystem.Threading.Tasks.TaskExceptionHolder\nVoid Finalize()",
  "HResult": -2146233088,
  "Source": "mscorlib",
  "WatsonBuckets": null,
  "InnerExceptions": [
    {
      "ClassName": "System.IO.FileNotFoundException",
      "Message": "Could not load file or assembly 'System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.",
      "Data": null,
      "InnerException": null,
      "HelpURL": null,
      "StackTraceString": "   at BorderlessGaming.BorderlessGaming.DoMainWork()\r\n   at System.Threading.Tasks.Task.Execute()",
      "RemoteStackTraceString": null,
      "RemoteStackIndex": 0,
      "ExceptionMethod": "8\nDoMainWork\nBorderlessGaming, Version=8.4.33315.1328, Culture=neutral, PublicKeyToken=null\nBorderlessGaming.BorderlessGaming\nVoid DoMainWork()",
      "HResult": -2147024894,
      "Source": "BorderlessGaming",
      "WatsonBuckets": null,
      "FileNotFound_FileName": "System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
      "FileNotFound_FusionLog": "WRN: Assembly binding logging is turned OFF.\r\nTo enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog](DWORD) to 1.\r\nNote: There is some performance penalty associated with assembly bind failure logging.\r\nTo turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].\r\n"
    }
  ]
}

{
  "ClassName": "System.AggregateException",
  "Message": "A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread.",
  "Data": null,
  "InnerException": {
    "ClassName": "System.IO.FileNotFoundException",
    "Message": "Could not load file or assembly 'System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.",
    "Data": null,
    "InnerException": null,
    "HelpURL": null,
    "StackTraceString": "   at BorderlessGaming.BorderlessGaming.DoMainWork()\r\n   at System.Threading.Tasks.Task.Execute()",
    "RemoteStackTraceString": null,
    "RemoteStackIndex": 0,
    "ExceptionMethod": "8\nDoMainWork\nBorderlessGaming, Version=8.4.33315.1328, Culture=neutral, PublicKeyToken=null\nBorderlessGaming.BorderlessGaming\nVoid DoMainWork()",
    "HResult": -2147024894,
    "Source": "BorderlessGaming",
    "WatsonBuckets": null,
    "FileNotFound_FileName": "System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
    "FileNotFound_FusionLog": "WRN: Assembly binding logging is turned OFF.\r\nTo enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog](DWORD) to 1.\r\nNote: There is some performance penalty associated with assembly bind failure logging.\r\nTo turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].\r\n"
  },
  "HelpURL": null,
  "StackTraceString": "   at System.Threading.Tasks.TaskExceptionHolder.Finalize()",
  "RemoteStackTraceString": null,
  "RemoteStackIndex": 0,
  "ExceptionMethod": "8\nFinalize\nmscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089\nSystem.Threading.Tasks.TaskExceptionHolder\nVoid Finalize()",
  "HResult": -2146233088,
  "Source": "mscorlib",
  "WatsonBuckets": null,
  "InnerExceptions": [
    {
      "ClassName": "System.IO.FileNotFoundException",
      "Message": "Could not load file or assembly 'System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.",
      "Data": null,
      "InnerException": null,
      "HelpURL": null,
      "StackTraceString": "   at BorderlessGaming.BorderlessGaming.DoMainWork()\r\n   at System.Threading.Tasks.Task.Execute()",
      "RemoteStackTraceString": null,
      "RemoteStackIndex": 0,
      "ExceptionMethod": "8\nDoMainWork\nBorderlessGaming, Version=8.4.33315.1328, Culture=neutral, PublicKeyToken=null\nBorderlessGaming.BorderlessGaming\nVoid DoMainWork()",
      "HResult": -2147024894,
      "Source": "BorderlessGaming",
      "WatsonBuckets": null,
      "FileNotFound_FileName": "System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
      "FileNotFound_FusionLog": "WRN: Assembly binding logging is turned OFF.\r\nTo enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog](DWORD) to 1.\r\nNote: There is some performance penalty associated with assembly bind failure logging.\r\nTo turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].\r\n"
    }
  ]
}

{
  "ClassName": "System.AggregateException",
  "Message": "A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread.",
  "Data": null,
  "InnerException": {
    "ClassName": "System.IO.FileNotFoundException",
    "Message": "Could not load file or assembly 'System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.",
    "Data": null,
    "InnerException": null,
    "HelpURL": null,
    "StackTraceString": "   at BorderlessGaming.BorderlessGaming.DoMainWork()\r\n   at System.Threading.Tasks.Task.Execute()",
    "RemoteStackTraceString": null,
    "RemoteStackIndex": 0,
    "ExceptionMethod": "8\nDoMainWork\nBorderlessGaming, Version=8.4.33315.1328, Culture=neutral, PublicKeyToken=null\nBorderlessGaming.BorderlessGaming\nVoid DoMainWork()",
    "HResult": -2147024894,
    "Source": "BorderlessGaming",
    "WatsonBuckets": null,
    "FileNotFound_FileName": "System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
    "FileNotFound_FusionLog": "WRN: Assembly binding logging is turned OFF.\r\nTo enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog](DWORD) to 1.\r\nNote: There is some performance penalty associated with assembly bind failure logging.\r\nTo turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].\r\n"
  },
  "HelpURL": null,
  "StackTraceString": "   at System.Threading.Tasks.TaskExceptionHolder.Finalize()",
  "RemoteStackTraceString": null,
  "RemoteStackIndex": 0,
  "ExceptionMethod": "8\nFinalize\nmscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089\nSystem.Threading.Tasks.TaskExceptionHolder\nVoid Finalize()",
  "HResult": -2146233088,
  "Source": "mscorlib",
  "WatsonBuckets": null,
  "InnerExceptions": [
    {
      "ClassName": "System.IO.FileNotFoundException",
      "Message": "Could not load file or assembly 'System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.",
      "Data": null,
      "InnerException": null,
      "HelpURL": null,
      "StackTraceString": "   at BorderlessGaming.BorderlessGaming.DoMainWork()\r\n   at System.Threading.Tasks.Task.Execute()",
      "RemoteStackTraceString": null,
      "RemoteStackIndex": 0,
      "ExceptionMethod": "8\nDoMainWork\nBorderlessGaming, Version=8.4.33315.1328, Culture=neutral, PublicKeyToken=null\nBorderlessGaming.BorderlessGaming\nVoid DoMainWork()",
      "HResult": -2147024894,
      "Source": "BorderlessGaming",
      "WatsonBuckets": null,
      "FileNotFound_FileName": "System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
      "FileNotFound_FusionLog": "WRN: Assembly binding logging is turned OFF.\r\nTo enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog](DWORD) to 1.\r\nNote: There is some performance penalty associated with assembly bind failure logging.\r\nTo turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].\r\n"
    }
  ]
}

{
  "ClassName": "System.AggregateException",
  "Message": "A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread.",
  "Data": null,
  "InnerException": {
    "ClassName": "System.IO.FileNotFoundException",
    "Message": "Could not load file or assembly 'System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.",
    "Data": null,
    "InnerException": null,
    "HelpURL": null,
    "StackTraceString": "   at BorderlessGaming.BorderlessGaming.DoMainWork()\r\n   at System.Threading.Tasks.Task.Execute()",
    "RemoteStackTraceString": null,
    "RemoteStackIndex": 0,
    "ExceptionMethod": "8\nDoMainWork\nBorderlessGaming, Version=8.4.33315.1328, Culture=neutral, PublicKeyToken=null\nBorderlessGaming.BorderlessGaming\nVoid DoMainWork()",
    "HResult": -2147024894,
    "Source": "BorderlessGaming",
    "WatsonBuckets": null,
    "FileNotFound_FileName": "System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
    "FileNotFound_FusionLog": "WRN: Assembly binding logging is turned OFF.\r\nTo enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog](DWORD) to 1.\r\nNote: There is some performance penalty associated with assembly bind failure logging.\r\nTo turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].\r\n"
  },
  "HelpURL": null,
  "StackTraceString": "   at System.Threading.Tasks.TaskExceptionHolder.Finalize()",
  "RemoteStackTraceString": null,
  "RemoteStackIndex": 0,
  "ExceptionMethod": "8\nFinalize\nmscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089\nSystem.Threading.Tasks.TaskExceptionHolder\nVoid Finalize()",
  "HResult": -2146233088,
  "Source": "mscorlib",
  "WatsonBuckets": null,
  "InnerExceptions": [
    {
      "ClassName": "System.IO.FileNotFoundException",
      "Message": "Could not load file or assembly 'System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.",
      "Data": null,
      "InnerException": null,
      "HelpURL": null,
      "StackTraceString": "   at BorderlessGaming.BorderlessGaming.DoMainWork()\r\n   at System.Threading.Tasks.Task.Execute()",
      "RemoteStackTraceString": null,
      "RemoteStackIndex": 0,
      "ExceptionMethod": "8\nDoMainWork\nBorderlessGaming, Version=8.4.33315.1328, Culture=neutral, PublicKeyToken=null\nBorderlessGaming.BorderlessGaming\nVoid DoMainWork()",
      "HResult": -2147024894,
      "Source": "BorderlessGaming",
      "WatsonBuckets": null,
      "FileNotFound_FileName": "System.Threading.Tasks, Version=1.5.11.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
      "FileNotFound_FusionLog": "WRN: Assembly binding logging is turned OFF.\r\nTo enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog](DWORD) to 1.\r\nNote: There is some performance penalty associated with assembly bind failure logging.\r\nTo turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].\r\n"
    }
  ]
}

Any idea what the issue is?

**Comment by psouza4** [comment]
You are using an outdated version that does not include the correct files,
or possibly do not have the latest .Net version installed.  Duplicate of
issue #219, if my memory serves... on mobile currently.

**Comment by Abulph** [comment]
Oh, I see. So I'll need .Net framework 4.5/4.6, not just 4?

**Comment by psouza4** [comment]
Yep.
